clusters/app.ci/prow/03_deployment/gcsweb.yaml

```diff
@@ -0,0 +1,195 @@
+apiVersion: v1
+kind: Template
+labels:
+  template: gcsweb
+metadata:
+  name: gcsweb
+  openshift.io/display-name: GCS Web UI
+  iconClass: icon-storage
+  tags: instant-app,storage
+  template.openshift.io/long-description: |-
+    The GCS Web interface is a web frontend to Google Cloud Storage that uses
+    the public, no-login-required API. Obviously this means it can only browse
+    public buckets.
+parameters:
+- name: NAME
+  displayName: Application Name
+  description: The name to use for the objects created by this template
+  value: gcsweb
+  required: true
+- name: SOURCE_REF
+  displayName: Source Ref
+  description: Git reference to use for building the GCS Web UI image
+  value: master
+  required: false
+- name: SOURCE_URL
+  displayName: Source URL
+  description: Git source URL to use for building the GCS Web UI image
+  value: https://github.com/openshift/test-infra.git
+  required: true
+- name: SOURCE_CONTEXT
+  displayName: Source Context
+  description: Git source context to use for building the GCS Web UI image
+  value: gcsweb
+  required: false
+- name: SERVICE_NAME
+  displayName: GCS Web UI Service Name
+  description: The name of the OpenShift Service exposed for the GCS Web UI container.
+  value: gcsweb
+- description: "The namespace. Must come from a var or it will get stripped off"
+  name: NAMESPACE
+  value: ci
+objects:
+- kind: ImageStream
+  apiVersion: v1
+  metadata:
+    labels:
+      app: "${NAME}"
+    name: "${NAME}"
+    namespace: ${NAMESPACE}
+- kind: BuildConfig
+  apiVersion: v1
+  metadata:
+    labels:
+      app: "${NAME}"
+    name: "${NAME}-binary"
+    namespace: ${NAMESPACE}
+  spec:
+    successfulBuildsHistoryLimit: 1
+    failedBuildsHistoryLimit: 2
+    output:
+      to:
+        kind: ImageStreamTag
+        name: "${NAME}:binaries"
+    source:
+      git:
+        ref: "${SOURCE_REF}"
+        uri: "${SOURCE_URL}"
+      contextDir: "${SOURCE_CONTEXT}"
+      type: Git
+      dockerfile: |
```

Comment:
Would it be hard to consume from upstream instead ot propagating this?

Reply:
Upstream neither has automated image publishing nor a manifest for gcsweb in their repo.

Reply:
Using the `prow_image` stuff in Bazel is pretty straightforward -- could we do that, please?

Reply:
yeah, but porting the Makefile is not. I've opened https://github.com/kubernetes/test-infra/issues/17043 for this.